core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/components/ai-image-prompt-form/ai-image-prompt-form.component.ts

```diff
@@ -0,0 +1,146 @@
+import { DropdownModule } from 'primeng/dropdown';
+import { InputTextareaModule } from 'primeng/inputtextarea';
+import { RadioButtonModule } from 'primeng/radiobutton';
+
+import { DotMessageService } from '@dotcms/data-access';
+import { DotFieldRequiredDirective, DotMessagePipe } from '@dotcms/ui';
+
+import { AIImageSize } from '../../../../shared';
+import {
+    AIImagePrompt,
+    DotGeneratedAIImage
+} from '../../../../shared/services/dot-ai/dot-ai.models';
+
+@Component({
+    selector: 'dot-ai-image-prompt-form',
+    standalone: true,
+    templateUrl: './ai-image-prompt-form.component.html',
+    imports: [
+        ButtonModule,
+        AccordionModule,
+        RadioButtonModule,
+        ReactiveFormsModule,
+        FormsModule,
+        DropdownModule,
+        NgIf,
+        InputTextareaModule,
+        DotFieldRequiredDirective,
+        DotMessagePipe
+    ],
+    styleUrls: ['./ai-image-prompt-form.component.scss']
+})
+export class AiImagePromptFormComponent implements OnChanges, OnInit {
+    @Input()
+    generatedValue: DotGeneratedAIImage;
+
+    @Input()
+    isLoading = false;
+
+    @Output()
+    value = new EventEmitter<AIImagePrompt>();
+
+    @Output()
+    orientation = new EventEmitter<AIImageSize>();
+
+    form: FormGroup;
+    aiProcessedPrompt: string;
+    dotMessageService = inject(DotMessageService);
+
+    orientationOptions: SelectItem<AIImageSize>[] = [
+        {
+            value: '1792x1024',
+            label: this.dotMessageService.get(
+                'block-editor.extension.ai-image.orientation.horizontal'
+            )
+        },
+        {
+            value: '1024x1024',
+            label: this.dotMessageService.get('block-editor.extension.ai-image.orientation.square')
+        },
+        {
+            value: '1024x1792',
+            label: this.dotMessageService.get(
+                'block-editor.extension.ai-image.orientation.vertical'
+            )
+        }
+    ];
+
+    ngOnInit(): void {
+        this.initForm();
+    }
+
+    /**
+     * Submits the form and emits the form value.
+     * @return {void}
+     * @memberof AiImagePromptFormComponent
+     */
+    submitForm(): void {
+        this.value.emit(this.form.value);
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
+        const { generatedValue, isLoading } = changes;
+        if (
+            generatedValue?.currentValue &&
+            !generatedValue.firstChange &&
+            !isLoading?.currentValue
+        ) {
+            const updatedValue: DotGeneratedAIImage = generatedValue.currentValue;
+            this.form.patchValue(updatedValue.request);
+            this.form.clearValidators();
+            this.form.updateValueAndValidity();
+
+            this.aiProcessedPrompt = updatedValue.response.revised_prompt;
+        }
+
+        if (isLoading && !isLoading.firstChange) {
+            isLoading.currentValue ? this.form.disable() : this.form.enable();
+        }
+    }
```

Comment:
An ngOnChanges can be replaced by input signals and computed signals. Only is an idea.